### ISSUE TITLE
Provide app for DSF members

### ISSUE THREAD

**Issue by freakboy3742** [django/djangoproject.com#362]
We need an app to capture information about DSF members (both corporate and developer)

Developer members have:
 - A name
 - An email address, 
 - A date when they joined
 - A date when they left
 - A reason why they left.

Corporate members have:
- Display name of company
- Formal name of company (billing purposes)
- Logo
- 1 paragraph description of company
- Company URL
- Initial contact date
- Membership start date
- Membership expiry date
- Membership contact email 
- Billing contact email
- Membership level
- Mailing address (international - not "US state and zip code")

Corporate data should be provided in a draft form by the user themselves, and then validated by a member of DSF staff. DSF staff should also be able to create a record on behalf of a company.

The two membership pages should then show the details for all current members, and provide a list of past members.

Nice to haves:
* Automated emails for key corporate events, like:
  - We've received your application
  - We've approved your application; invoice incoming
  - We've received your payment
  - We're still waiting on your payment
  - Your membership is about to expire
  - Your membership has expired.

 Logs of these contacts should also be included.

* Automated tie ins to Google groups membership - subscribing/unsubscribing member contacts from list.

* Allowing companies to update their own records.

**Comment by relekang** [comment]
[I started to work on this](https://github.com/django/djangoproject.com/compare/django:master...relekang:members-app). Will be doing more throughout the week.

**Comment by shaib** [comment]
The spec above is a little weird with respect to  membership history. It includes some historical data (membership start dates and, for corporates, even first-contact date) but cannot support any non-trivial history (member joins, leaves, re-joins -- which is their membership start date?; corporate changes level -- initial level forgotten?) or track payment history.

Also unspecified (and somewhat related to history): Once a company's details have been approved, the company should be able to update them -- I suppose such changes should, again, be submitted as a draft, and updated in the site once approved by DSF staff (this does not contradict the spec above, but the initial implementation in #504 does not seem to support it well)

**Comment by timgraham** [comment]
The basics of this are done. Enhancements can be ticketed out separately.
